pkg/apis/hco/v1beta1/hyperconverged_types.go

```diff
@@ -66,67 +69,73 @@ type HyperConvergedFeatureGates struct {
 	// This may degrade virt-launcher security.
 	// +optional
 	// +kubebuilder:default=false
-	SRIOVLiveMigration *bool `json:"sriovLiveMigration,omitempty"`
+	SRIOVLiveMigration FeatureGate `json:"sriovLiveMigration,omitempty"`
 
 	// Allow attaching a data volume to a running VMI
 	// +optional
 	// +kubebuilder:default=false
-	HotplugVolumes *bool `json:"hotplugVolumes,omitempty"`
+	HotplugVolumes FeatureGate `json:"hotplugVolumes,omitempty"`
 
 	// Allow assigning GPU and vGPU devices to virtual machines
 	// +optional
 	// +kubebuilder:default=false
-	GPU *bool `json:"gpu,omitempty"`
+	GPU FeatureGate `json:"gpu,omitempty"`
 
 	// Allow assigning host devices to virtual machines
 	// +optional
 	// +kubebuilder:default=false
-	HostDevices *bool `json:"hostDevices,omitempty"`
+	HostDevices FeatureGate `json:"hostDevices,omitempty"`
 
 	// Allow migrating a virtual machine with CPU host-passthrough mode. This should be
 	// enabled only when the Cluster is homogeneous from CPU HW perspective doc here
 	// +optional
 	// +kubebuilder:default=false
-	WithHostPassthroughCPU *bool `json:"withHostPassthroughCPU,omitempty"`
+	WithHostPassthroughCPU FeatureGate `json:"withHostPassthroughCPU,omitempty"`
 
 	// Support migration for VMs with host-model CPU mode
 	// +optional
 	// +kubebuilder:default=true
-	WithHostModelCPU *bool `json:"withHostModelCPU,omitempty"`
+	WithHostModelCPU FeatureGate `json:"withHostModelCPU,omitempty"`
 
 	// Enable HyperV strict host checking for HyperV enlightenments
 	// Defaults to true, even when HyperConvergedFeatureGates is empty
 	// +optional
 	// +kubebuilder:default=true
-	HypervStrictCheck *bool `json:"hypervStrictCheck,omitempty"`
-}
+	HypervStrictCheck FeatureGate `json:"hypervStrictCheck,omitempty"`
 
-func (fgs *HyperConvergedFeatureGates) IsHotplugVolumesEnabled() bool {
-	return (fgs != nil) && (fgs.HotplugVolumes != nil) && (*fgs.HotplugVolumes)
+	// +optional
+	// +k8s:openapi-gen=false
+	enabledGateList map[string]bool `json:"-"`
 }
 
-func (fgs *HyperConvergedFeatureGates) IsGPUAssignmentEnabled() bool {
-	return (fgs != nil) && (fgs.GPU != nil) && (*fgs.GPU)
-}
+func (fgs *HyperConvergedFeatureGates) RebuildEnabledGateMap() map[string]bool {
```

Comment:
Here we are caching values of other public fields in a private field. As far as I see, we have to call this method in some places to invalidate the cache. I think we should not do it. Cache invalidation is one of the hardest things in software development :) 

Also, I didn't get what we gain with this change. If code duplication is the matter, I think we can improve it with some helper functions.